### PR TITLE
Support ipv6 connect for #290 #293 #332 #470

### PR DESCRIPTION
**Pull request by buaazp**
Connect to ipv6 addr failed because url.host().to_string() got a `[fdbd:dc02:ff:1:1:174:229:45]` String.

Which cannot parse to IPv6Addr: 
https://doc.rust-lang.org/src/std/net/addr.rs.html#963-980

Just use IPv6Addr's to_string() aka Display trait can fix it.
https://doc.rust-lang.org/src/std/net/ip.rs.html#1758-1850

### CODE REVIEW

File: redis/Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "redis"
-version = "0.21.5"
+version = "0.21.6"
```
Comment (djc) [inline redis/Cargo.toml:3]: This shouldn't be a part of your PR, probably some rebase/merge issue?

File: redis/src/connection.rs [hunk 2]
```diff
@@ -167,7 +167,11 @@ impl IntoConnectionInfo for String {
 
 fn url_to_tcp_connection_info(url: url::Url) -> RedisResult<ConnectionInfo> {
     let host = match url.host() {
-        Some(host) => host.to_string(),
+        Some(host) => match host {
+            url::Host::Domain(path) => path.to_string(),
+            url::Host::Ipv4(v4) => v4.to_string(),
+            url::Host::Ipv6(v6) => v6.to_string(),
+        },
```
Comment (djc) [inline redis/src/connection.rs:174]: Maybe add a comment here on why we can't use `host.to_string()` directly, to preserve your research?

**Review by djc — APPROVED** [review]
Nice, thanks!

File: redis/Cargo.toml (hunk 1 above)
Comment (buaazp) [reply redis/Cargo.toml]: Got it. Removed.

File: redis/src/connection.rs (hunk 2 above)
Comment (buaazp) [reply redis/src/connection.rs]: Sure, this issue is quite convoluted and should indeed be explained with enough comments.

**Comment by buaazp** [comment]
Updated follow the review comments. And added some ipv6 test cases to make sure this change is OK.

**Review by djc — APPROVED** [review]

**Review by jaymell — APPROVED** [review]

**Pull request merged**